cloud/pkg/cloudhub/servers/httpserver/pre_server.go

```diff
@@ -159,32 +159,38 @@ func createCertsToSecret(ctx context.Context) error {
 
 // GenerateAndRefreshToken creates a token and save it to secret, then craete a timer to refresh the token.
 func GenerateAndRefreshToken(ctx context.Context) error {
-	caHashToken, err := token.Create(hubconfig.Config.Ca, hubconfig.Config.CaKey,
-		hubconfig.Config.CloudHub.TokenRefreshDuration)
-	if err != nil {
-		return fmt.Errorf("failed to generate the token for edgecore register, err: %v", err)
-	}
-	// save caHashAndToken to secret
-	if err := client.SaveSecret(ctx, createTokenSecret([]byte(caHashToken)), constants.SystemNamespace); err != nil {
-		return fmt.Errorf("failed to create tokenSecret, err: %v", err)
+	if err := createNewToken(ctx); err != nil {
+		return err
 	}
-
 	t := time.NewTicker(time.Hour * hubconfig.Config.CloudHub.TokenRefreshDuration)
 	go func() {
 		for {
 			select {
 			case <-t.C:
-				caHashToken, err = token.Create(hubconfig.Config.Ca, hubconfig.Config.CaKey,
-					hubconfig.Config.CloudHub.TokenRefreshDuration)
-				if err != nil {
-					klog.Errorf("failed to refresh the token for edgecore register, err: %v", err)
+				if err := createNewToken(ctx); err != nil {
+					klog.Warningf("failed to refresh the new token, err: %v", err)
+					return
 				}
+				klog.Info("token refreshed successfully")
 			case <-ctx.Done():
 				break
 			}
 		}
 	}()
-	klog.Info("Succeed to creating token")
+	klog.Info("token created successfully")
```

Comment:
Do we need to add log level here and L174?

Reply:
When the token is not refreshed, we can search the history log by information. The frequency of printing here is very low.

Reply:
Sounds good.